### PR TITLE
Bump to NLPModels 0.15

### PR DESCRIPTION
**Pull request by tmigot**
Follows [NLPModels.jl PR #353](https://github.com/JuliaSmoothOptimizers/NLPModels.jl/pull/353) with parametric meta.
The tests were breaking so I had to let go on 0.14.

### CODE REVIEW

File: src/stats.jl [hunk 1]
```diff
@@ -74,11 +74,10 @@ end
...
 function GenericExecutionStats(
   status::Symbol,
-  nlp::AbstractNLPModel;
-  solution::V = eltype(nlp.meta.x0)[],
+  nlp::AbstractNLPModel{T, S};
+  solution::V = T[],
   kwargs...,
-) where {V}
-  T = eltype(solution)
+) where {S, T, V}
```
Comment (abelsiqueira) [inline src/stats.jl:80]: I think `V` already serves the purpose of `S`.

Comment (tmigot) [reply src/stats.jl:80]: Coulnd't we have `nlp.meta.x0` of one type, and the solution of a solver of a different type (but with the same `eltype`) ?
For instance, `x0` is a `Vector{Float64}` and `solution` is a `SubArray{Float64, 1}` or `SparseVector{Float64}` ?

Comment (abelsiqueira) [reply src/stats.jl:80]: You're right. Now I understood that `S` is used only in the constructor.

**Comment by abelsiqueira** [comment]
Thanks

**Pull request merged**